topic_dynamics/parsing.py

```diff
@@ -0,0 +1,342 @@
+                    identifiers.append(identifier)
+            if len(child.children) != 0:
+                traverse_tree(child)
+
+    traverse_tree(root)
+    sorted_identifiers = sorted(Counter(identifiers).items(), key=itemgetter(1), reverse=True)
+
+    return sorted_identifiers
+
+
+def transform_identifiers(identifiers: List) -> List[str]:
+    """
+    Transform the original list of identifiers into the writable form.
+    :param identifiers: list of tuples, identifier and count.
+    :return: a list of identifiers in the writable form, "identifier:count".
+    """
+    formatted_identifiers = []
+    for identifier in identifiers:
+        if identifier[0].rstrip() != "":  # Checking for occurring empty tokens.
+            formatted_identifiers.append(identifier[0].rstrip() + ":" + str(identifier[1]).rstrip())
+    return formatted_identifiers
+
+
+def slice_and_parse(repository: str, n_dates: int, time_delta: int, lang: str, name: str) -> None:
+    """
+    Split the repository, parse the full files, write the data into a file.
+    Can be called for parsing full files and for parsing diffs only.
+    :param repository: path to the repository to process.
+    :param n_dates: the amount of dates.
+    :param time_delta: the time step between dates.
+    :param lang: language of parsing.
+    :param name: name of the dataset (directories with resulting files).
+    :return: None.
+    """
+    print("Creating the temporal slices of the data.")
+    # Create a folder for created files
+    directory = os.path.abspath(os.path.join(repository, os.pardir, name + "_processed"))
+    os.mkdir(directory)
+    dates = get_dates(n_dates, time_delta)
+    dates_indices = {}
+    count = 0
+    # Create temporal slices of the project, get a list of files for each slice, parse all files, save the tokens
+    with open(os.path.abspath(os.path.join(directory, name + "_tokens.txt")), "w+") as fout:
+        for date in tqdm(dates):
+            with TemporaryDirectory() as td:
+                subdirectory = os.path.abspath(os.path.join(directory, td, date.strftime("%Y-%m-%d")))
+                checkout_by_date(repository, subdirectory, date)
+                files = get_files(subdirectory, get_extensions(lang))
+                starting_index = count + 1
+                for file in files:
+                    if os.path.isfile(file):  # TODO: implement a better file-checking mechanism
+                        try:
+                            identifiers = get_identifiers(file, lang)
+                            if len(identifiers) != 0:
+                                count += 1
+                                formatted_identifiers = transform_identifiers(identifiers)
+                                fout.write(str(count) + ";" + os.path.relpath(file, os.path.abspath(os.path.join(directory, td))) + ";" + ",".join(formatted_identifiers) + "\n")
```

Comment:
Create class with overrided [`repr`](https://stackoverflow.com/questions/8494358/defining-repr-when-subclassing-set-in-python)  to store this information - it will be much easier to understand what you are writing in future

Reply:
I created this class, but I currently fail to implement it. I'm not sure how to replace fout(...) with it. And also, there might be more than "a" and "b", something like *args? Could you please provide an example of using this so that I can implement it everywhere?